app/src/main/java/com/github/lookupgroup27/lookup/model/map/renderables/utils/GeometryUtils.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.lookupgroup27.lookup.model.map.renderables.utils
+
+/**
+ * Utility object for generating spherical geometry. This class provides methods for creating vertex
+ * data, color data, texture coordinates, and indices for rendering a 3D sphere using OpenGL.
+ *
+ * The sphere is tessellated into latitude bands and longitude steps, which control the level of
+ * detail and smoothness of the sphere.
+ */
+object Sphere {
```

Comment:
Would it make sense to rename this object to `GeometryUtils`? It might better reflect its purpose and align with our naming conventions.